Library/Homebrew/compilers.rb

```diff
@@ -6,6 +6,7 @@ module CompilerConstants
     "gcc-4.0"  => :gcc_4_0,
     "gcc-4.2"  => :gcc,
     "clang"    => :clang,
+    "brew_clang" => :brew_clang,
```

Comment:
That'll change every release and I think we don't actually need to bump it each version like we do with GCC. I think ideally we'd figure out a way eventually to have this just be `:clang` and `:gcc` and specify the build/version inside the DSL if necessary.